test/e2e/framework/dump.go

```diff
@@ -0,0 +1,214 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+func retrieveContainerLogs(ctx context.Context, podClient corev1client.PodInterface, destinationPath string, podName string, logOptions *corev1.PodLogOptions) error {
+	dest, err := os.OpenFile(destinationPath, os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := dest.Close()
+		if err != nil {
+			klog.ErrorS(err, "can't close file", "Path", destinationPath)
+		}
+	}()
+
+	logsReq := podClient.GetLogs(podName, logOptions)
+	readCloser, err := logsReq.Stream(ctx)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := readCloser.Close()
+		if err != nil {
+			klog.ErrorS(err, "can't close log stream", "Path", destinationPath, "Pod", podName, "Container", logOptions.Container)
+		}
+	}()
+
+	_, err = io.Copy(dest, readCloser)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func DumpResource(ctx context.Context, dynamicClient dynamic.Interface, coreClient corev1client.CoreV1Interface, gvr schema.GroupVersionResource, groupDir, namespace string) error {
+	list, err := dynamicClient.Resource(gvr).Namespace(namespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(list.Items) == 0 {
+		return nil
+	}
+
+	for _, obj := range list.Items {
+		obj.SetManagedFields(nil)
+	}
+
+	resourceDir := filepath.Join(groupDir, gvr.Resource)
+	err = os.Mkdir(resourceDir, 0777)
+	if err != nil {
+		return err
+	}
+
+	listData, err := yaml.Marshal(list.Items)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(filepath.Join(groupDir, fmt.Sprintf("%s.yaml", gvr.Resource)), listData, 0666)
+	if err != nil {
+		return err
+	}
+
+	for _, obj := range list.Items {
+		data, err := yaml.Marshal(obj)
+		if err != nil {
+			return err
+		}
+
+		err = ioutil.WriteFile(filepath.Join(resourceDir, fmt.Sprintf("%s.yaml", obj.GetName())), data, 0666)
+		if err != nil {
+			return err
+		}
+
+		switch gvr.String() {
+		case corev1.SchemeGroupVersion.WithResource("pods").String():
```

Comment:
nit: case body is too long, extracting it to DumpPod would reduce lenght of entire function and indentation

Reply:
done